mesonbuild/mintro.py

```diff
@@ -125,14 +125,15 @@ def list_installed(installdata: backends.InstallData) -> T.Dict[str, str]:
             res[basename] = os.path.join(installdata.prefix, s.install_path, basename)
     return res
 
-def list_install_plan(installdata: backends.InstallData) -> T.Dict[str, T.Dict[str, T.Dict[str, T.Optional[str]]]]:
-    plan: T.Dict[str, T.Dict[str, T.Dict[str, T.Optional[str]]]] = {
+def list_install_plan(installdata: backends.InstallData) -> T.Dict[str, T.Dict[str, T.Dict[str, T.Optional[str] | T.List[str]]]]:
+    plan: T.Dict[str, T.Dict[str, T.Dict[str, T.Optional[str] | T.List[str]]]] = {
         'targets': {
             Path(installdata.build_dir, target.fname).as_posix(): {
                 'destination': target.out_name,
                 'tag': target.tag or None,
                 'subproject': target.subproject or None,
-                'install_rpath': target.install_rpath or None
+                'install_rpath': target.install_rpath or None,
+                'build_rpaths': sorted(x.decode('utf8') for x in target.rpath_dirs_to_remove),
```

Comment:
For reasons, `target.rpath_dirs_to_remove` is kept as a list of byte strings. I'm not sure decoding those as UTF8 is always the right thing to do, but it is done elsewhere in the codebase, and I cannot come up with something more correct.